src/fsharp/FSharp.Compiler.LanguageServer/LspTypes.fs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.LanguageServer
+
+// Types as defined at https://microsoft.github.io/language-server-protocol/specification
+
+/// Represents a zero-based line and column of a text document.
+type Position = {
+    line: int
```

Comment:
Nit: here and elsewhere - upper case labels.

Reply:
I purposefully left these lower case to match the exact LSP interface member to make JSON serialization/deserialization easier.  Otherwise we have to annotate every field:

``` F#
type Diagnostic =
    {
        [<JsonProperty("range")>]
        Range: Range
        [<JsonProperty("severity")>]
        Severity: int
        [<JsonProperty("code")>]
        Code: string
        [<JsonProperty("source")>]
        Source: string
        [<JsonProperty("message")>]
        Message: string
        [<JsonProperty("relatedInformation")>]
        RelatedInformation: DiagnosticRelatedInformation[]
    } with
    static member Error = 1
    static member Warning = 2
    static member Information = 3
    static member Hint = 4
```

If you feel strongly about upper case then I can change it, but if you're indifferent I'd rather leave them lower case to make the structures easier to read.

Reply:
Definitely indifferent give this 😄

Reply:
Maybe a quick comment near the top of the file explaining this?